public/docs/_examples/cb-validation/ts/app/hero-form-model-binding.component.html

```diff
@@ -0,0 +1,75 @@
+ <!-- #docplaster -->
+ <!-- #docregion -->
+<div class="container">
+  <div [hidden]="submitted">
+    <h1>Hero Form (Model-Driven with Data Binding)</h1>
+    <form [ngFormModel]="heroForm" *ngIf="active" 
+          (ngSubmit)="onSubmit()">
+      <div class="form-group">
+        <!-- #docregion name-with-error-msg -->
+        <label for="name">Name</label>
+        <input type="text" class="form-control" 
+               [(ngModel)]="model.name"
+               ngControl="name"
+               [ngClass]="{'required': isRequired('name')}">
```

Comment:
Glad you found the accessibility issues, but the entire point of model-driven is that the knowledge of the validation is *only* in the code. That way the validation can be adjusted based on the user or on the state (different validation for add vs edit for example). So we don't want to put the validation attributes into the HTML in the model-driven cases. I'm putting together an issue for Angular Forms to suggest that aria attributes are automatically added to the HTML based on the code.